### DISCUSSION TITLE
Producing a confidence interval for a percentile estimate (not the median)

### DISCUSSION THREAD

**Discussion by LukeAFullard** [scikit-learn-contrib/MAPIE#293] [Q&A] [answered]
Hello,
Please excuse me if this turns out to be an ignorant question!

I wish to predict a specific percentile of time-series data over time and put a confidence interval on my prediction.
For example, if we had housing data, I may like to predict the 90th percentile (p90) of home sale prices over time, and then I may like to add a 95% confidence interval around my prediction p90.

In the examples I have read with MAPIE so far it seems that people are predicting the median and adding a 90% prediction interval by calculating the 5th and 95th percentiles of the data. This seems different to what I am trying to do (build a confidence interval around a specific percentile of the data).

Is what I am trying to do reasonable? Possible with MAPIE?

Thank you for your time, and bearing with my probably basic question :)

**Comment by thibaultcordier** [comment, marked as answer]
Helo @LukeAFullard,
Not at all, your question is interesting and makes sense! Your request is clear and you understand how MAPIE for TimeSeries works.

I think what you are trying to do is not possible with MAPIE. Indeed, with conformal prediction framework, we are looking for this statement: $p[Y \in C(X)] \leq 1-\alpha$ (see [this gentle introduction](https://arxiv.org/abs/2107.07511) for more details). In other words, we are trying to find a $1-\alpha$ confidence interval around the mediane prediction $\hat Y$.

If I rephrase your proposal, you are looking for: $p[Q(Y,p) \in C(X)] \leq 1-\alpha$ where $Q(Y,p)$ corresponds to the p-th percentile of the probabilistic distribution of the random variable $Y$. In other words, you are trying to find a $1-\alpha$ confidence interval around your quantile prediction $\hat Q(Y,p)$. But you need to know the data distribution of $Q(Y,p)$ (or have a data set of this distribution) and you only have access to the $Y$ distribution (and not $Q(Y,p)$ ). In this situation, the conformal prediction framework cannot help you directly.
 
I hope my answer will help you find a solution to your problem :)

**Reply by LukeAFullard** [reply]
That is clear, thank you very much for your reply :)
